cmd/object-handlers.go

```diff
@@ -1517,7 +1524,10 @@ func (api objectAPIHandlers) NewMultipartUploadHandler(w http.ResponseWriter, r
 		return
 	}
 
-	retPerms := isPutActionAllowed(getRequestAuthType(r), bucket, object, r, iampolicy.PutObjectRetentionAction)
+	var retPerms APIErrorCode
+	if r.Header.Get(xhttp.AmzObjectLockMode) != "" || r.Header.Get(xhttp.AmzObjectLockRetainUntilDate) != "" {
+		retPerms = isPutActionAllowed(getRequestAuthType(r), bucket, object, r, iampolicy.PutObjectRetentionAction)
```

Comment:
rather than repeating it here, can we not do this inside the `isPutActionAllowed` function when action == iampolicy.PutObjectRetentionAction

Reply:
done